tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_gpu.h

```diff
@@ -92,12 +115,319 @@ class TableWrapper final : public TableWrapperBase<K, V> {
+
+    FileSystem* fs;
+    const auto env = ctx->env();
+    TF_RETURN_WITH_CONTEXT_IF_ERROR(
+        env->GetFileSystemForFile(filepath, &fs),
+        "Please make sure you have already imported tensorflow_io before using "
+        "TFRA file system operation.");
+
+    const string key_filepath = filepath + "-keys";
+    TF_RETURN_IF_ERROR(fs->FileExists(key_filepath));
+    std::unique_ptr<RandomAccessFile> key_file;
+    TF_RETURN_IF_ERROR(fs->NewRandomAccessFile(key_filepath, &key_file));
+    std::unique_ptr<io::RandomAccessInputStream> key_input_stream(
+        new io::RandomAccessInputStream(key_file.get()));
+    const size_t key_buffer_byte_size = buffer_size * sizeof(K);
+    io::BufferedInputStream key_reader(key_input_stream.get(),
+                                       key_buffer_byte_size * 2);
+
+    const string value_filepath = filepath + "-values";
+    TF_RETURN_IF_ERROR(fs->FileExists(value_filepath));
+    std::unique_ptr<RandomAccessFile> value_file;
+    TF_RETURN_IF_ERROR(fs->NewRandomAccessFile(value_filepath, &value_file));
+    std::unique_ptr<io::RandomAccessInputStream> value_input_stream(
+        new io::RandomAccessInputStream(value_file.get()));
+    const size_t value_len = sizeof(V) * dim;
+    const size_t value_buffer_byte_size = buffer_size * value_len;
+    io::BufferedInputStream value_reader(value_input_stream.get(),
+                                         value_buffer_byte_size * 2);
+
+    uint64 key_file_size = 0;
+    TF_RETURN_IF_ERROR(fs->GetFileSize(key_filepath, &key_file_size));
+    const size_t key_size = key_file_size / sizeof(K);
+
+    uint64 value_file_size = 0;
+    TF_RETURN_IF_ERROR(fs->GetFileSize(value_filepath, &value_file_size));
+    const size_t value_size = value_file_size / value_len;
+
+    if (key_size != value_size) {
+      return errors::Unavailable(
+          "the keys number in file " + key_filepath +
+          " is not equal to the value vectors number in file " +
+          value_filepath + ".");
+    }
+
+    // Rehash table
+    size_t last_size_hint = table_->get_size(stream);
+    size_t table_max_size = table_->get_capacity();
+    this->rehash_if_needed(&last_size_hint, &table_max_size, 0, stream,
+                           key_size);
+
+    K* d_keys = nullptr;
+    V* d_values = nullptr;
+    CUDA_CHECK(cudaMallocAsync(&d_keys, key_buffer_byte_size, stream));
+    CUDA_CHECK(cudaMallocAsync(&d_values, value_buffer_byte_size, stream));
+#define CLEANUP_CUDA_CODE                    \
+  CUDA_CHECK(cudaFreeAsync(d_keys, stream)); \
+  CUDA_CHECK(cudaFreeAsync(d_values, stream));
+
+    tstring key_buffer;
+    key_buffer.resize(key_buffer_byte_size);
+    tstring value_buffer;
+    value_buffer.resize(value_buffer_byte_size);
+
+    size_t key_file_offset = 0;
+    size_t remainder = key_file_size - key_file_offset;
+    size_t nkeys = 0;
+    size_t key_read_byte = 0;
+    size_t value_read_byte = 0;
+    while (remainder > 0) {
```

Comment:
`remainder` (size_t, unsigned) is always greater than or equal to `0`. It is better to use `int64`.